lib/iris/tests/unit/fileformats/netcdf/test__load_aux_factory.py

```diff
@@ -110,13 +110,38 @@ def test_formula_terms_p0_bounded(self):
             self.assertEqual(msg, str(warn[0].message))
 
     def test_formula_terms_ap_missing_coords(self):
-        coordinates = [(mock.sentinel.b, 'b'), (mock.sentinel.ps, 'ps')]
-        self.provides = dict(coordinates=coordinates)
         self.requires['formula_terms'] = dict(ap='ap', b='b', ps='ps')
         with mock.patch('warnings.warn') as warn:
             _load_aux_factory(self.engine, self.cube)
         warn.assert_called_once_with("Unable to find coordinate for variable "
                                      "'ap'")
+        # Check cube.add_aux_coord method.
+        self.assertEqual(self.cube.add_aux_coord.call_count, 0)
+        # Check cube.add_aux_factory method.
+        self.assertEqual(self.cube.add_aux_factory.call_count, 1)
+        args, _ = self.cube.add_aux_factory.call_args
+        self.assertEqual(len(args), 1)
+        factory = args[0]
+        # Check that the factory has no delta term
+        self.assertEqual(factory.delta, None)
+        self.assertEqual(factory.sigma, mock.sentinel.b)
+        self.assertEqual(factory.surface_air_pressure, mock.sentinel.ps)
+
+    def test_formula_terms_no_delta_terms(self):
+        self.requires['formula_terms'] = dict(b='b', ps='ps')
```

Comment:
Yes, I think it's possible, but nonsensical. Since the two terms are multiplied together in [the formula](http://cfconventions.org/cf-conventions/v1.6.0/cf-conventions.html#_atmosphere_hybrid_sigma_pressure_coordinate), if one of them is zero it doesn't matter what the other one is, so there'd be no point having both. I'll add tests anyway though.